ttnn/cpp/ttnn/operations/reduction/moe/device/moe_program_factory.hpp

```diff
@@ -1,15 +1,33 @@
-// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+// SPDX-FileCopyrightText: © 2025 Tenstorrent AI ULC
 //
 // SPDX-License-Identifier: Apache-2.0
 
-#include "ttnn/run_operation.hpp"
+#pragma once
 
-namespace ttnn::operations::reduction::detail {
+#include "ttnn/device_operation.hpp"
+#include "moe_device_operation_types.hpp"
 
-tt::tt_metal::operation::ProgramWithCallbacks moe_single_core_interleaved(
-    const Tensor& input_tensor,
-    const Tensor& expert_mask_tensor,
-    const Tensor& topk_mask_tensor,
-    uint16_t k,
-    Tensor& out_tensor);
-}
+namespace ttnn::operations::reduction::moe::program {
+
+struct MoeSharedVariables {
+    tt::tt_metal::KernelHandle unary_reader_kernel_id;
+    tt::tt_metal::KernelHandle unary_writer_kernel_id;
```

Comment:
clang tidy on ci will tell that primitive types must be initialized to avoid issues
```suggestion
    tt::tt_metal::KernelHandle unary_reader_kernel_id {};
    tt::tt_metal::KernelHandle unary_writer_kernel_id {};
```